From 84cf4a9f54e71da29f7919f1a1549d941ff67e8d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Bernon?= <rbernon@codeweavers.com>
Date: Fri, 13 Dec 2024 19:30:51 +0100
Subject: [PATCH] winex11: Move GL/VK offscreen if the clipping region is
 NULLREGION.

NULLREGION means that the entire window should be clipped out, and its
children displayed instead. The change broke some offscreening decisions
when both parent and children have a GL drawable created but drawing
actually happens on the child window.

Fixes: 786d9d1685ac220081b10cc779d4d331ddd2fc52
Wine-Bug: https://bugs.winehq.org/show_bug.cgi?id=57503
---
 dlls/winex11.drv/init.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/winex11.drv/init.c b/dlls/winex11.drv/init.c
index b8184e59bb1..274ed51f585 100644
--- a/dlls/winex11.drv/init.c
+++ b/dlls/winex11.drv/init.c
@@ -215,7 +215,7 @@ static BOOL needs_client_window_clipping( HWND hwnd )
     if ((region = NtGdiCreateRectRgn( 0, 0, 0, 0 )))
     {
         ret = NtGdiGetRandomRgn( hdc, region, SYSRGN | NTGDI_RGN_MONITOR_DPI );
-        if (ret > 0 && (ret = NtGdiGetRgnBox( region, &rect )) <= NULLREGION) ret = 0;
+        if (ret > 0 && (ret = NtGdiGetRgnBox( region, &rect )) < NULLREGION) ret = 0;
         if (ret == SIMPLEREGION && EqualRect( &rect, &client )) ret = 0;
         NtGdiDeleteObjectApp( region );
     }
-- 
GitLab

